Wallet and testapp release builds won't run

Both `:wallet` and `:samples:testapp` fail to run when compiled with release build types (`upstreamRelease` and `blueRelease`, respectively) with the following error:

```
Process: org.multipaz.androidwallet, PID: 6478
java.lang.RuntimeException: Unable to create application org.multipaz.wallet.WalletApplication: java.lang.IllegalStateException: Error occurred verifying signature
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7003)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.IllegalStateException: Error occurred verifying signature
	at sa.b.a(Unknown Source:186)
	at qa.c.a(Unknown Source:125)
	at org.multipaz.wallet.WalletApplication.onCreate(Unknown Source:757)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6998)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: java.lang.IllegalStateException: Unexpected error
	at sa.i.h(Unknown Source:124)
	at sa.b.a(Unknown Source:156)
	at qa.c.a(Unknown Source:125) 
	at org.multipaz.wallet.WalletApplication.onCreate(Unknown Source:757) 
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316) 
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6998) 
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: java.security.NoSuchAlgorithmException: no such algorithm: EC for provider BC
	at sun.security.jca.GetInstance.getService(GetInstance.java:87)
	at sun.security.jca.GetInstance.getInstance(GetInstance.java:206)
	at java.security.Security.getImpl(Security.java:631)
	at java.security.AlgorithmParameters.getInstance(AlgorithmParameters.java:303)
	at sa.i.h(Unknown Source:53)
	at sa.b.a(Unknown Source:156) 
	at qa.c.a(Unknown Source:125) 
	at org.multipaz.wallet.WalletApplication.onCreate(Unknown Source:757) 
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316) 
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6998) 
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
```

These are caused by R8 stripping the BouncyCastle BC provider and EC algorithm, because it looks like they're not being used. Obviously they are, but the way they are used through Java security provider, R8 doesn't see it. This can be fixed by adding necessary keep rules to the `:multipaz` `consumer-rules.pro` file, e.g.:

```
-keep class org.bouncycastle.jcajce.provider.keystore.BC$Mappings { *; }
-keep class org.bouncycastle.jcajce.provider.asymmetric.EC$Mappings { *; }
-keep class org.bouncycastle.jcajce.provider.keystore.bc.* { *; }
-keep class org.bouncycastle.jcajce.provider.asymmetric.ec.* { *; }
```

I can provide a PR that fixes this issue.

--- a/multipaz/build.gradle.kts
+++ b/multipaz/build.gradle.kts
@@ -209,6 +209,7 @@ android {
     defaultConfig {
         minSdk = 26
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles("consumer-rules.pro")
     }
 
     compileOptions {

--- a/multipaz/consumer-rules.pro
+++ b/multipaz/consumer-rules.pro
@@ -1,0 +1,5 @@
+# Prevent BouncyCastle BC provider and EC algorithm from being stripped out
+-keep class org.bouncycastle.jcajce.provider.keystore.BC$Mappings { *; }
+-keep class org.bouncycastle.jcajce.provider.asymmetric.EC$Mappings { *; }
+-keep class org.bouncycastle.jcajce.provider.keystore.bc.* { *; }
+-keep class org.bouncycastle.jcajce.provider.asymmetric.ec.* { *; }

